src/main/java/com/shapesecurity/shift/es2018/parser/GenericParser.java

```diff
@@ -1723,62 +1763,115 @@ protected Expression parseNewExpression() throws JsError {
             if (!(callee instanceof Expression)) {
                 throw this.createUnexpected(this.lookahead);
             }
-            return this.finishNode(startState, new NewExpression((Expression) callee, this.match(TokenType.LPAREN) ? this.parseArgumentList().left : ImmutableList.empty()));
+            return this.finishNode(startState, new NewExpression((Expression) callee, this.match(TokenType.LPAREN) ? this.parseArgumentList() : ImmutableList.empty()));
         } else if (fromParseLeftHandSideExpression.isMiddle()) {
             throw this.createUnexpected(this.lookahead);
         } else {
             throw this.createError(ErrorMessages.UNEXPECTED_OBJECT_BINDING);
         }
     }
 
+    @Nonnull
+    protected ImmutableList<SpreadElementExpression> parseArgumentList() throws JsError {
+        Either<Pair<ImmutableList<SpreadElementExpression>, Maybe<SourceLocation>>, FormalParameters> args = this.parseArgumentListOrAsyncArrowParameters();
+        if (args.isRight()) {
+            assert this.firstExprError != null;
+            throw this.firstExprError;
+        }
+        return args.left().fromJust().left;
+    }
+
     /**
-     * parse argument list
+     * parse argument list, including surrounding parentheses
+     *
+     * @return either: a pair of `SpreadElementExpression` and potentially the SourceLocation after the first `SpreadElement`
+     *          or, if one of the "arguments" can only be parsed as an assignment target, a FormalParameters node (used for async arrow expressions)
      *
-     * @return a pair of `SpreadElementExpression` and potentially the SourceLocation after the first `SpreadElement`.
+     *          // Either<Pair<ImmutableList<SpreadElementExpression>, Maybe<SourceLocation>>, FormalParameters>
```

Comment:
This line is should get replaced with something like

"If this can be parsed as an argument list, returns a pair representing those arguments and potentially the SourceLocation after the first `SpreadElement`. Otherwise, if this can only be parsed as the formal parameters of an async arrow, returns a `FormalParameters` node holding them.